src/pages/Earn/useFarmRegistry.ts

```diff
@@ -89,15 +107,60 @@ export const useFarmRegistry = () => {
           tvlUSD: farmData[e.returnValues.stakingAddress].tvlUSD,
           rewardsUSDPerYear: farmData[e.returnValues.stakingAddress].rewardsUSDPerYear,
           isFeatured: !!featuredPoolWhitelist[e.returnValues.stakingAddress],
+          rewardApr: undefined,
+          swapApr: undefined,
+          apr: undefined,
+          apy: undefined,
         })
       })
 
     farmSummaries
       .sort((a, b) => Number(fromWei(toBN(b.rewardsUSDPerYear).sub(toBN(a.rewardsUSDPerYear)))))
       .sort((a, b) => Number(fromWei(toBN(b.tvlUSD).sub(toBN(a.tvlUSD)))))
 
-    setFarmSummaries(farmSummaries)
-  }, [kit.web3.eth])
+    const results = await Promise.all(
+      farmSummaries.map((summary) => {
+        return client.query({ query: pairDataGql, variables: { id: summary.lpAddress.toLowerCase() } })
+      })
+    )
+    const farmInfos = results.map((result: ApolloQueryResult<any>, index) => {
+      let swapRewardsUSDPerYear = 0
+      const summary = farmSummaries[index]
+      const { loading, error, data } = result
+      if (!loading && !error && data) {
+        const lastDayVolumeUsd = data.pair.pairHourData.reduce(
+          (acc: number, curr: { hourlyVolumeUSD: string }) => acc + Number(curr.hourlyVolumeUSD),
+          0
+        )
+        swapRewardsUSDPerYear = Math.floor(lastDayVolumeUsd * 365 * 0.0025)
+      }
+      const rewardApr = new Percent(summary.rewardsUSDPerYear, summary.tvlUSD)
+      const swapApr = new Percent(toWei(swapRewardsUSDPerYear.toString()), summary.tvlUSD)
+      const apr = new Percent(
+        toBN(toWei(swapRewardsUSDPerYear.toString())).add(toBN(summary.rewardsUSDPerYear)).toString(),
+        summary.tvlUSD
+      )
+      let apy: string | undefined = undefined
+      try {
+        apy = annualizedPercentageYield(apr, COMPOUNDS_PER_YEAR)
+      } catch (e) {
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        console.error('apy calc overflow', summary.farmName, e)
+      }
+      return {
+        rewardApr,
+        swapApr,
+        apr,
+        apy,
+      }
+    })
+    setFarmSummaries(
+      farmSummaries.map((summary, index) => ({
+        ...summary,
+        ...farmInfos[index],
```

Comment:
why not just do all this calculation in L64? Then you don't need to additionally set the values and initialize those fields to `undefined`

Reply:
why was this resolved? Is it possible to push the values here:

```
farmSummaries.push({
          farmName: ethers.utils.parseBytes32String(e.returnValues.farmName),
          stakingAddress: e.returnValues.stakingAddress,
          lpAddress: e.returnValues.lpAddress,
          token0Address: lps[e.returnValues.lpAddress][0],
          token1Address: lps[e.returnValues.lpAddress][1],
          tvlUSD: farmData[e.returnValues.stakingAddress].tvlUSD,
          rewardsUSDPerYear: farmData[e.returnValues.stakingAddress].rewardsUSDPerYear,
          isFeatured: !!featuredPoolWhitelist[e.returnValues.stakingAddress],
+rewardApr,
+swapApr,
+apr,
+apy
        })
```